client/src/main/java/org/asynchttpclient/netty/request/NettyRequestSender.java

```diff
@@ -332,7 +326,7 @@ protected void onFailure(Throwable cause) {
         Uri uri = request.getUri();
         final Promise<List<InetSocketAddress>> promise = ImmediateEventExecutor.INSTANCE.newPromise();
 
-        if (proxy != null && !proxy.isIgnoredForHost(uri.getHost())) {
+        if (proxy != null && !proxy.isIgnoredForHost(uri.getHost()) && !proxy.getProxyType().isSocks()) {
```

Comment:
for consistency, should be proxyServer.getProxyType() == ProxyType.HTTP

Reply:
Actually, is this really needed? The point here is to honor proxy's ignoredHosts, doesn't seem related to proxy type.